internal/command/machine/run.go

```diff
@@ -426,35 +426,83 @@ func determineServices(ctx context.Context) ([]api.MachineService, error) {
 			handlers = append(handlers, splittedProtoHandlers[1:]...)
 		}
 
-		splittedPorts := strings.Split(splittedPortsProto[0], ":")
-		edgePort, err := strconv.Atoi(splittedPorts[0])
+		edgePort, edgeStartPort, edgeEndPort, internalPort, err := parsePorts(splittedPortsProto[0])
 		if err != nil {
-			return nil, errors.Wrap(err, "invalid edge port")
-		}
-		machinePort := edgePort
-		if len(splittedPorts) > 1 {
-			machinePort, err = strconv.Atoi(splittedPorts[1])
-			if err != nil {
-				return nil, errors.Wrap(err, "invalid machine (internal) port")
-			}
+			return nil, err
 		}
 
-		edgePort32 := int32(edgePort)
-
 		machineServices[i] = api.MachineService{
 			Protocol:     proto,
-			InternalPort: machinePort,
+			InternalPort: internalPort,
 			Ports: []api.MachinePort{
 				{
-					Port:     &edgePort32,
-					Handlers: handlers,
+					Port:      edgePort,
+					StartPort: edgeStartPort,
+					EndPort:   edgeEndPort,
+					Handlers:  handlers,
 				},
 			},
 		}
 	}
 	return machineServices, nil
 }
 
+func parsePorts(input string) (port, start_port, end_port *int32, internal_port int, err error) {
+	split := strings.Split(input, ":")
+	if len(split) == 1 {
+		var external_port int
+		external_port, err = strconv.Atoi(split[0])
+		if err != nil {
+			err = errors.Wrap(err, "invalid port")
+			return
+		}
+
+		p := int32(external_port)
+		port = &p
+	} else {
```

Comment:
It looks like we might want to sanity check that `split` doesn't have more than two items? For instance `80:8080:3000` is ambiguous and doesn't mean anything.